rgw/v2/tests/s3_swift/reusables/rgw_s3_elbencho.py

```diff
@@ -0,0 +1,276 @@
+import json
+import logging
+import os
+import random
+import re
+import subprocess
+import threading
+import time
+
+import boto
+import boto3
+import v2.utils.utils as utils
+from v2.lib.exceptions import SyncFailedError, TestExecError
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.write_io_info import (
+    AddUserInfo,
+    BasicIOInfoStructure,
+    BucketIoInfo,
+    IOInfoInitialize,
+    KeyIoInfo,
+)
+from v2.tests.s3_swift import reusable
+from v2.tests.s3_swift.reusables import rgw_accounts as accounts
+
+log = logging.getLogger()
+
+
+def json_serial(obj):
+    """JSON serializer for objects not serializable by default json code."""
+    if isinstance(obj, datetime):
+        return obj.isoformat()
+    raise TypeError(f"Type {type(obj)} not serializable")
+
+
+def get_endpoint_elbencho():
+    """Determines the appropriate endpoint for Elbencho based on HAProxy availability."""
+    log.info("Checking HAProxy status")
+    try:
+        haproxy_status = utils.exec_shell_cmd("systemctl is-active haproxy")
+        if haproxy_status.strip() == "active":
+            log.info("HAProxy is active, retrieving hostname")
+            hostname = utils.exec_shell_cmd("hostname -s").strip()
+            return f"http://{hostname}:5000"
+    except Exception as e:
+        log.error(f"Failed to determine HAProxy status: {e}")
+    return None
+
+
+def get_remote_endpoint_elbencho():
+    """Fetches the HAProxy-enabled hostname on the remote site."""
+    try:
+        remote_site_ssh_con = reusable.get_remote_conn_in_multisite()
+        stdin, stdout, stderr = remote_site_ssh_con.exec_command(
+            "sudo ceph orch host ls"
+        )
+        cmd_output = stdout.read().decode()
+
+        log.info(f"Remote site host list:\n{cmd_output}")
+
+        for line in cmd_output.split("\n"):
+            if "ha_io" in line:
```

Comment:
@viduship 
just an enhancement we can have in coming PR's
to filter the hosts based on label:

[cephuser@ceph-pri-hsm-scale-w9woqm-node6 ~]$ ceph orch host ls --label rgw --format json-pretty

[
  {
    "addr": "10.0.67.100",
    "hostname": "ceph-pri-hsm-scale-w9woqm-node5",
    "labels": [
      "rgw",
      "osd"
    ],
    "status": ""
  }
]
[cephuser@ceph-pri-hsm-scale-w9woqm-node6 ~]$

Reply:
but we need ha_io in order to get the ha endpoint for io